### PR TITLE
Fixed failed to delete function when function's file is not exist

### PR DESCRIPTION
**Pull request by life1347**
This PR fixed the problem that user may encounter when trying to delete a function without function file in filestore. Now, the controller checks the file existence before removing it. (https://github.com/fission/fission/issues/108)

```
$ fission fn list
NAME  UID                                  ENV
hello bdecc45c-a528-4f1f-9fca-728bdc3d86f6 nodejs
$ fission fn delete --name hello
Failed to delete function 'hello': Delete failed: remove /filestore/bdecc45c-a528-4f1f-9fca-728bdc3d86f6: no such file or directory
```

### CODE REVIEW

**Comment by life1347** [comment]
Also, we should return more meaningful error message back to user when trying to get function file that not exist. (https://github.com/fission/fission/issues/112)

```
$ fission function get --name hello
Failed to get function: (Error 0) HTTP error 500
```

**Comment by tobias** [comment]
With #116, that message gets a little better:

    $ fission fn get --name foo
    Failed to get function: (Error 0) 100: Key not found (/Function/foo) [17]

but #116 doesn't map the etcd 100 error to fission's ErrorNotFound - I'll do that in a later PR if #116 is accepted. Thanks for pointing it out!

File: controller/fileStore.go [hunk 1]
```diff
@@ -85,7 +85,11 @@ func (fs *FileStore) fileStoreService() {
 		case WRITE:
 			response.error = ioutil.WriteFile(path.Join(fs.root, req.fileName), req.fileContents, 0600)
 		case DELETE:
-			response.error = os.Remove(path.Join(fs.root, req.fileName))
+			response.error = nil
+			filePath := path.Join(fs.root, req.fileName)
+			if _, err := os.Stat(filePath); os.IsExist(err) {
```
Comment (soamvasani) [inline controller/fileStore.go]: Won't err just be nil if the stat succeeds?  I would suggest unconditionally doing the `os.Remove` but checking for and ignoring err when `os.IsNotExist(err)`.

Comment (life1347) [inline controller/fileStore.go]: @soamvasani thanks for your suggestion! I just updated the commit. By the way, should I squash the commits into one?

Comment (soamvasani) [inline controller/fileStore.go]: Thanks for the update! No need, I can squash the commits.

**Comment by soamvasani** [comment]
BTW, #108 is about error handing in creating the filestore path itself; that's a bit different from the bug addressed here.

**Pull request merged**